privacyidea/lib/policydecorators.py

```diff
@@ -296,10 +296,9 @@ def auth_user_passthru(wrapped_function, user_object, passw, options=None):
     from privacyidea.lib.token import get_tokens
     options = options or {}
     g = options.get("g")
-    if get_tokens(user=user_object, count=True) == 0 and g:
```

Comment:
I am wondering if we could get a logical problem here!

The policy is logically only checked, if the user has no token.

If the policy check would raise an exception, maybe due to conflicting policies, we would change the behaviour of the system. Even a user *with* a valid token could not login, since  a policy, that has no relevance for him, would raise an exception.

I *think* in this case there will be no policy exception, since we do not use *unique=True*, but this is something we need to consider carefully. What do you think?

Reply:
Ah, didn't think of that! Yes, we should make sure that we don't change behavior in the case you describe. Policy conflicts are only checked if the user has no tokens (``check_for_conflicts`` in line 313), but I'll check if ``get_policies`` can throw any other exceptions.

Reply:
Having thought about this some more: ``get_policies`` should normally not throw any exceptions, and conflict checking is only done if the user has, in fact, zero tokens. We use ``get_policies`` in a lot of other policy functions without checking for errors, so I think that if the call goes wrong here, this would be a hint that something is seriously wrong with the privacyIDEA instance, and having an error would be sensible in that case. What do you think?